src/components/Search.jsx

```diff
@@ -0,0 +1,165 @@
+  import.meta.env.PUBLIC_ALGOLIA_APP_ID,
+  import.meta.env.PUBLIC_ALGOLIA_API_KEY
+)
+
+function Search() {
+  const [isSearchVisible, setIsSearchVisible] = useState(false)
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debounceQuery, setDebounceQuery] = useState('')
+  const debounceTimeoutRef = useRef(null)
+
+  useEffect(() => {
+    const toggleSearch = () => {
+      setIsSearchVisible(prev => {
+        if (prev) {
+          setSearchQuery('')
+        }
+        return !prev
+      })
+    }
+
+    const handleKeyDown = event => {
+      if ((event.metaKey || event.ctrlKey) && event.key === 'k') {
+        event.preventDefault()
+        toggleSearch()
+      }
+    }
+
+    const handleSearchClick = event => {
+      if (event.target.closest('.search-click')) {
+        toggleSearch()
+      }
+    }
+
+    document.addEventListener('keydown', handleKeyDown)
+    document.addEventListener('click', handleSearchClick)
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown)
+      document.removeEventListener('click', handleSearchClick)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (isSearchVisible && searchQuery) {
+      document.body.classList.add('no-scroll')
+    } else {
+      document.body.classList.remove('no-scroll')
+    }
+  }, [isSearchVisible, searchQuery])
+
+  useEffect(() => {
+    if (debounceTimeoutRef.current) {
+      clearTimeout(debounceTimeoutRef.current)
+    }
+
+    debounceTimeoutRef.current = setTimeout(() => {
+      setDebounceQuery(searchQuery)
+    }, 400)
+
+    return () => {
+      if (debounceTimeoutRef.current) {
+        clearTimeout(debounceTimeoutRef.current)
+      }
+    }
+  }, [searchQuery])
+
+  return (
+    <>
+      {isSearchVisible && (
+        <div id="searchbox-wrapper" className="searchbox-wrapper">
+          <InstantSearch indexName="docs" searchClient={searchClient}>
+            <SearchBox
+              translations={{ placeholder: 'Search daytona.io' }}
+              autoFocus={true}
+              onChange={event => setSearchQuery(event.currentTarget.value)}
+              value={searchQuery}
+            />
+            {debounceQuery && (
+              <>
+                <div
+                  id="stats-pagination-wrapper"
+                  className="stats-pagination-wrapper"
+                >
+                  <Stats />
+                  <Pagination
+                    showFirst={false}
+                    showPrevious={true}
+                    showNext={true}
+                    showLast={false}
+                    padding={1}
+                  />
+                </div>
+                <Hits hitComponent={Hit} />
+              </>
+            )}
+            <Configure
+              hitsPerPage={10}
+              clickAnalytics={true}
+              getRankingInfo={false}
+            />
+          </InstantSearch>
+        </div>
+      )}
+    </>
+  )
+}
+
+function Hit({ hit }) {
```

Comment:
![Screenshot 2024-11-08 at 08 56 45](https://github.com/user-attachments/assets/694900c9-214f-4c99-8bdb-1bae80fce4cf)

The focus outline of the Hit is a bit janky. Make it so that only the outline of the container is show (if possible).